rpm/generic/zfs-dkms.spec.in

```diff
@@ -32,6 +32,13 @@ Obsoletes:      spl-dkms
 Provides:       %{module}-kmod = %{version}
 AutoReqProv:    no
 
+%if (0 < 0%{?fedora} && 0%{?fedora} <= 30) || (0 < 0%{?rhel} && 0%{?rhel} <= 9) || (0 < 0%{?centos} && 0%{?centos} <= 9)
+# We don't need it, but if this isn't installed, rpmbuild as root can
+# crash+corrupt rpmdb
+# See issue #12071
+BuildRequires:  ncompress
+%endif
```

Comment:
Is there any harm in making this an unconditional dependency?  I hate having extra dependencies, but in this case it seems like perhaps the least bad solution.  It's one more package to add to the documentation but sounds like we'll want to do that either way.

Reply:
Well, actual unconditional breaks builds on deb-based systems; assuming you meant shoving it under the same `%if 0%{?rhel}%{?fedora}%{?suse_version}` above, I have no real objection, I was just trying to avoid unnecessarily enforcing that constraint; if we're going to make it "unconditional" in the RPM spec, then I should just drop the configure check, since the only difference becomes it dying slightly earlier.

RPM supports expressing something like BuildRequires: (file >= 5.37 or ncompress), but only starting in rpm 4.13, and since EL7 uses 4.11.x, that seems like a nonstarter.

Reply:
And pushed. I tested it on a deb and rpm based system, so I hope I didn't leave any hot tar bubbling in a corner.

Reply:
Yeah, I think that's all right.  And it looks like things correctly failed because of the new missing dependency, so it'll just need to be added to the CI and docs.  Thanks for filing this upstream, perhaps they'll sort it out.  I've had better luck reporting things against CentOS Stream 8 for what it's worth.